wallet/src/key_chain/mod.rs

```diff
@@ -0,0 +1,366 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # BIP39 key chain
+//! The KeyChain struct holds and constantly derives keys for the wallet addresses
+//! It uses the following derivation scheme:
+//!
+//! m/44'/19788'/<account_number>'/<key_purpose>'/<key_index>'
```

Comment:
The comment says that `key_purpose` and `key_index` are hardened keys, but in the code they are normal keys.